build/debian-base/Makefile

```diff
@@ -27,6 +27,8 @@ ALL_ARCH = amd64 arm arm64 ppc64le s390x
 TEMP_DIR:=$(shell mktemp -d)
 QEMUVERSION=v2.9.1
 
+SUDO=$(if $(filter 0,$(shell id -u)),,sudo)
```

Comment:
To explain this: `make` doesn't let you do equality checks in variable assignment, so this is next best option. Basically, `filter pattern,text` returns whitespace-separated words from `text` that match the `pattern`, so here we're returning `0` when the user is root and empty string otherwise.

the `if` in variable assignment considers nonempty strings as true, empty strings as false. thus when the effective user id is 0 (i.e. you're root), `filter` returns `0`, which evaluates to true, and so we don't use `sudo`. Otherwise we do.

Reply:
(this implementation was basically copied from https://stackoverflow.com/a/9008922/9723266.)